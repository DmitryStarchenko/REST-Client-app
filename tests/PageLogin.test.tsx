import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, it, expect, vi } from 'vitest';

vi.mock('@/components/auth/LoginForm/LoginForm', () => ({
  default: vi.fn(() => <div data-testid="login-form">Login Form</div>),
}));

import Login from '../src/app/[locale]/(auth)/login/page';

describe('Login Page', () => {
  it('should render without crashing', () => {
    render(<Login />);
    expect(screen.getByTestId('login-form')).toBeInTheDocument();
  });

  it('should render LoginForm component', () => {
    render(<Login />);

    const loginForm = screen.getByTestId('login-form');
    expect(loginForm).toBeInTheDocument();
    expect(loginForm).toHaveTextContent('Login Form');
  });

  it('should match snapshot', () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });
});
