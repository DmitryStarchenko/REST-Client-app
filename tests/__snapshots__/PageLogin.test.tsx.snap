// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page > should match snapshot 1`] = `
<div>
  <div
    data-testid="login-form"
  >
    Login Form
  </div>
</div>
`;
